Signal/src/ViewControllers/ConversationView/ConversationViewItem.m

```diff
@@ -466,51 +471,80 @@ - (nullable TSAttachmentPointer *)attachmentPointer
     return _attachmentPointer;
 }
 
-- (CGSize)contentSize
+- (CGSize)mediaSize
 {
     OWSAssertIsOnMainThread();
     OWSAssert(self.hasViewState);
 
-    return _contentSize;
+    return _mediaSize;
 }
 
 #pragma mark - UIMenuController
 
-- (NSArray<UIMenuItem *> *)menuControllerItems
+- (NSArray<UIMenuItem *> *)textMenuControllerItems
 {
     return @[
+        [[UIMenuItem alloc] initWithTitle:NSLocalizedString(@"EDIT_ITEM_COPY_ACTION",
+                                              @"Short name for edit menu item to copy contents of media message.")
+                                   action:self.copyTextActionSelector],
         [[UIMenuItem alloc] initWithTitle:NSLocalizedString(@"EDIT_ITEM_SHARE_ACTION",
                                               @"Short name for edit menu item to share contents of media message.")
-                                   action:self.shareActionSelector],
+                                   action:self.shareTextActionSelector],
+        [[UIMenuItem alloc] initWithTitle:NSLocalizedString(@"EDIT_ITEM_MESSAGE_METADATA_ACTION",
+                                              @"Short name for edit menu item to show message metadata.")
+                                   action:self.metadataActionSelector],
+        // FIXME: when deleting a caption, users will be surprised that it also deletes the attachment.
+        // We either need to implement a way to remove the caption separate from the attachment
+        // or make a design change which clarifies that the whole message is getting deleted.
```

Comment:
Is there a card?

Reply:
Another argument for WA-style UI which visually groups this content.

It's a shame that we can't make a long-press on hybrid messages to show "copy image" "copy caption", etc.